### PR TITLE
Refactor: Stop renaming/relabeling in each draw.

### PR DESCRIPTION
**Pull request by karupayun**
Before this commit we renamed each track every time when we
draw them. Now, we are making that only once.

### CODE REVIEW

File: OrbitGl/TimeGraph.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ TimeGraph::TimeGraph() : batcher_(BatcherId::kTimeGraph) {
       GetOrCreateThreadTrack(TracepointEventBuffer::kAllTracepointsFakeTid);
 
   // The process track is a special ThreadTrack of id "kAllThreadsFakeTid".
-  process_track_ = GetOrCreateThreadTrack(SamplingProfiler::kAllThreadsFakeTid);
+  process_track_ = nullptr;
```
Comment (florian-kuebler) [inline OrbitGl/TimeGraph.cpp:45]: I though about this again, and that member is only used in the SortTracks method. So I would remove it from the class, and have it as a local variable in SortTracks

File: OrbitGl/TimeGraph.cpp [hunk 2]
```diff
@@ -319,6 +319,9 @@ void TimeGraph::ProcessTimer(const TimerInfo& timer_info, const FunctionInfo* fu
     } else {
       scheduler_track_->OnTimer(timer_info);
       cores_seen_.insert(timer_info.processor());
+      uint32_t num_cores = GetNumCores();
```
Comment (florian-kuebler) [inline OrbitGl/TimeGraph.cpp:322]: I don't think that you should set the name/number of cores everytime you process a scheduling timer. This should be probably done in the GetOrCreateSchedulerTrack

**Review by florian-kuebler — APPROVED** [review]
Despite the two minor comments, LGTM. Thanks!

File: OrbitGl/TimeGraph.cpp (hunk 1 above)
Comment (karupayun) [reply OrbitGl/TimeGraph.cpp:45]: Yes, you are right!

File: OrbitGl/TimeGraph.cpp (hunk 2 above)
Comment (karupayun) [reply OrbitGl/TimeGraph.cpp]: The little issue here is that we don't know the NumCores before processing all timers, do we?

cores_seen_ is simply a set of cores (in my case 8) and we are inserting each processor several times, I changed it, but not sure if you will like it

**Review by pierricgimmig — APPROVED** [review]
Thanks for following up on this Pablo! LGTM.

File: OrbitGl/TimeGraph.cpp [hunk 3]
```diff
@@ -318,7 +312,12 @@ void TimeGraph::ProcessTimer(const TimerInfo& timer_info, const FunctionInfo* fu
       ++thread_count_map_[timer_info.thread_id()];
     } else {
       scheduler_track_->OnTimer(timer_info);
-      cores_seen_.insert(timer_info.processor());
+      if (!cores_seen_.count(timer_info.processor())) {
```
Comment (florian-kuebler) [inline OrbitGl/TimeGraph.cpp:315]: I feel that this cores_seen_ is a bit unintuitive. Couldn't we just have a `num_cores_` variable, and here check `if(num_cores_ < timer_info.processor()) num_cores_ = timer_info.processor(); ....`? 
But anyways. Other than that, it looks good. Also this is just a nit. Feel free to ignore.

**Review by florian-kuebler — APPROVED** [review]
Despite the nit, LGTM. Thanks!

File: OrbitGl/TimeGraph.cpp (hunk 3 above)
Comment (karupayun) [reply OrbitGl/TimeGraph.cpp:315]: Oh, I didn't suppose that the processors went from 0 to N-1. Is it a guarantee?

Comment (florian-kuebler) [reply OrbitGl/TimeGraph.cpp:315]: Yes, it is.

**Pull request merged**